testsuite/pytests/sli2py_connect/test_common_properties_setting.py

```diff
@@ -0,0 +1,135 @@
+    "stdp_pl_synapse_hom": {"parameter": "tau_plus", "value": 10, "setup": None, "neuron": "iaf_psc_alpha"},
+    "stdp_synapse_hom": {"parameter": "tau_plus", "value": 10, "setup": None, "neuron": "iaf_psc_alpha"},
+    "static_synapse_hom_w": {"parameter": "weight", "value": 10, "setup": None, "neuron": "iaf_psc_alpha"},
+    "tsodyks_synapse_hom": {"parameter": "tau_psc", "value": 10, "setup": None, "neuron": "iaf_psc_alpha"},
+}
+
+# Filter models that may not be built in
+available_cp_models = {k: v for k, v in common_prop_models.items() if k in nest.synapse_models}
+
+
+@pytest.fixture(autouse=True)
+def reset_kernel():
+    nest.ResetKernel()
+
+
+@pytest.mark.parametrize("syn_model, specs", available_cp_models.items())
+def test_set_common_properties(syn_model, specs):
+    """Test that setting a parameter works"""
+
+    if specs["setup"]:
+        specs["setup"]()
+
+    old_val = nest.GetDefaults(syn_model)[specs["parameter"]]
+    assert old_val != specs["value"]  # test would be meaningless otherwise
+
+    nest.SetDefaults(syn_model, {specs["parameter"]: specs["value"]})
+    new_val = nest.GetDefaults(syn_model)[specs["parameter"]]
+    assert new_val == specs["value"]
+
+
+@pytest.mark.parametrize("syn_model, specs", available_cp_models.items())
+def test_copy_common_properties(syn_model, specs):
+    """Test that parameters set on a model are copied and that setting on the copy does not touch the original"""
+
+    if specs["setup"]:
+        specs["setup"]()
+
+    old_val = nest.GetDefaults(syn_model)[specs["parameter"]]
+    assert old_val != specs["value"]  # test would be meaningless otherwise
+
+    nest.SetDefaults(syn_model, {specs["parameter"]: specs["value"]})
+    new_model = syn_model + "_copy"
+    nest.CopyModel(syn_model, new_model)
+    new_val = nest.GetDefaults(new_model)[specs["parameter"]]
+    assert new_val == specs["value"]
+
+    # Set parameter back on copied model, original must not be changed
+    nest.SetDefaults(new_model, {specs["parameter"]: old_val})
+    assert nest.GetDefaults(syn_model)[specs["parameter"]] == specs["value"]
+
+
+@pytest.mark.parametrize("syn_model, specs", available_cp_models.items())
+def test_no_setting_on_connection(syn_model, specs):
+    """Test that common property cannot be set on individual connection."""
+
+    if specs["setup"]:
+        specs["setup"]()
+
+    n = nest.Create(specs["neuron"])
+    nest.Connect(n, n, syn_spec={"synapse_model": syn_model})
+    conn = nest.GetConnections()
+    with pytest.raises(nest.kernel.NESTError):
+        conn.set({specs["parameter"]: specs["value"]})
+
+
+@pytest.mark.parametrize("syn_model, specs", available_cp_models.items())
+def test_no_setting_on_connect(syn_model, specs):
+    """Test that common property cannot be set in a Connect call"""
+
+    if specs["setup"]:
+        specs["setup"]()
+
+    n = nest.Create(specs["neuron"])
+    with pytest.raises(nest.kernel.NESTError):
```

Comment:
Specific error?

Reply:
Fixed. Here we get the NotImplemented from the checking function.